src/main/java/nc/students/ayaz/repositories/UserRepository.java

```diff
@@ -0,0 +1,26 @@
+package nc.students.ayaz.repositories;
+
+import nc.students.ayaz.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+public class UserRepository {
+
+    private Map<String, User> users = new HashMap<>();
```

Comment:
And we are going to iterate whole set. It looks a little bit complicated. Seems its quite easier to use HashMap